custom-stats-logger/src/main/java/dev/gemfire/stat/logger/StatLogger.java

```diff
@@ -55,14 +55,20 @@ public void execute(FunctionContext functionContext) {
       return;
     }
 
-    long timerInterval = Long.parseLong(logFrequency);
-    createTimerRegion(functionContext.getCache());
+    long timerInterval;
     try {
+      timerInterval = Long.parseLong(logFrequency);
+      createTimerRegion(functionContext.getCache());
       List<StatsHolder> statsHolderList = createStatsList(functionContext.getCache().getDistributedSystem(), statistics);
       createTimer(functionContext.getCache(), statsHolderList, timerInterval);
       functionContext.getResultSender().lastResult("Logging Metric count " + statsHolderList.size() + " with timer interval set to " + timerInterval + " ms");
     }
+    catch (NumberFormatException exception) {
+      cancelTimer(functionContext.getCache());
+      functionContext.getResultSender().sendException(exception);
+    }
```

Comment:
Unless `createTimerRegion()`, `createStatsList()` and `createTimer()` can also throw a `NumberFormatException`, the try/catch should just be around the call to `parseLong()`, with the function returning after sending the exception, with no need to cancel the timer, because we wouldn't have created the region or the timer task when the exception was thrown.